### ISSUE TITLE
Import from Thingiverse/Printables

### ISSUE THREAD

**Issue by aneurinprice** [manyfold3d/manyfold#1990]
**Describe the solution you'd like**
I use manyfold for offline storage of externally published models. A REALLY nice feature would be the ability to paste in a url to Thingiverse or Printables and have the model imported automagically. Ideally the description and licence would be populated and it would pull additional images that are not included in the files/images path.

**Describe alternatives you've considered**
Manually copying and pasting this data :(

**Comment by Floppy** [comment]
Yes yes yes yes. I would love to do this. It's definitely on my one-day list.

**Comment by aneurinprice** [comment]
I think it would also be cool to have Manyfold sync from thingiverse collections/likes.

**Comment by Floppy** [comment]
Absolutely - that was one of my original ideas, back a long time ago before I started writing them down here :)

**Comment by ns-mglaske** [comment]
+1 for this option!  Also, maybe check for updates, although this would require versioning of models.  Additionally, possibly import from Github ?

**Comment by aneurinprice** [comment]
Gitlab for 3d models actually sounds amazing.

**Comment by Floppy** [comment]
automatically grabbing updated files is definitely on the wishlist for this, yes 👍🏻

**Comment by Virtualchronos** [comment]
> Gitlab for 3d models actually sounds amazing.

Nice indeed but way too heavy in ressources to be a valid solution. Manyfold even if not perfect is way more adapted

**Comment by FallenLeave91** [comment]
Maker World would also be nice.

**Comment by fidoriel** [comment]
> Maker World would also be nice.

also myminifactory and thangs (the stuff hosted there)

**Comment by Floppy** [comment]
Anything with API, really!

**Comment by madcar86** [comment]
+1

**Comment by TheMBeat** [comment]
Maybe a useful lib: https://github.com/makerbot/thingiverse-ruby

**Comment by Floppy** [comment]
Good spot, thanks! We'll use that if we can (and bring it up to date if necessary)

**Comment by diedrichg** [comment]
Excellent idea and exactly what I feel would make offline storage quick and helpful.

**Comment by gbomacfly** [comment]
Would appreciate :)

**Comment by Floppy** [comment]
Didn't mean to close this yet!

**Comment by Floppy** [comment]
Good news, with all the sync work that's been happening, for Thingiverse I can now do this on the dev console in one line, so we're nearly there!

`Library.default.models.create(name: "4062607", path: "4062607", links_attributes: [{url: "https
://www.thingiverse.com/thing:4062607"}]).links.first.update_metadata_from_link_later`

As far as doing this from the UI, I think I might add something to the search controller so if you enter a URL into the search box, it will give you the option to import it.
